example.py

```diff
@@ -106,14 +106,14 @@ def main(
 
 cheese =>""",
     ]
-    results = generator.generate(
-        prompts, max_gen_len=256, temperature=temperature, top_p=top_p
-    )
-
-    for result in results:
-        print(result)
-        print("\n==================================\n")
-
+    for idx in range(0, len(prompts), max_batch_size):
+        actual_prompts = prompts[idx:idx+max_batch_size]
```

Comment:
I think that the only thing that would be needed to be handled is in the case where we set a max_batch_size higher than the amount of prompts:
```python
prompts[idx:min(idx+max_batch_size, len(prompts))]
```
If not we would be getting an index error.

Reply:
I dont know. This is not necessary because python handle this.  I search and python's docs say ""The slice of s from i to j is defined as the sequence of items with index k such that i <= k < j. If i or j is greater than len(s), use len(s). If i is omitted or None, use 0. If j is omitted or None, use len(s). If i is greater than or equal to j, the slice is empty." for s[i:j]  you can check yourself by [this link](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range)

Reply:
yes true! Very nice then! Thankss